crates/shared/src/transport/http.rs

```diff
@@ -147,6 +147,9 @@ impl BatchTransport for HttpTransport {
 
         async move {
             let _guard = metrics.on_request_start("batch");
+            for call in &calls {
```

Comment:
Based on the docs for `scopeguard` this should not work as expected:
https://docs.rs/scopeguard/latest/scopeguard/

So, as I understand, the point here is to increment the counter of `inflight` before executing the request, and then decrement the `inflight` and increment the `completed` after executing the request. 

With your code, I would expect for all of this to happen inside the inner scope.

Seems to me that we actually need to change the signature of the `on_request_start` function to receive a slice of methods (not only one method), and then handle that change inside `on_request_start`

Reply:
I liked that calling `on_request_start()` a bunch of times is easy to implement but changing the signature to receive an iterator of labels is probably harder to misuse and would allow us to write the code such that only a single timer needs to be created per batch call. 🤔